### ISSUE TITLE
net::ERR_NAME_NOT_RESOLVED

### ISSUE THREAD

**Issue by MarcGodard** [socketio/socket.io#5172]
**Describe the bug**

https://pagespeed.web.dev/analysis/https-rapiddevgroup-com/uzk899s2pg?form_factor=desktop

Error when using pagespeed insights

![image](https://github.com/user-attachments/assets/cb396c6c-67ca-46d6-bb0a-685451f5c3a4)

*Server*

"version": "4.7.5",

*Client*

"version": "4.7.5",

```
    const socket = io(apiSocket, {
      transports: ['websocket'],
      upgrade: true,
      query: { websiteId }
    })
```

This might be a limitation of pagespeed, of a setting I am using above. Considering I didn't see a lot of other people having this issue, it might be my settings, or a bug.

**Comment by darrachequesne** [comment]
Hi! A `ERR_NAME_NOT_RESOLVED` error means the DNS request for the `api.rapiddevgroup.com` domain failed.

Which is weird, since it seems to be resolvable:

```
$ dig api.rapiddevgroup.com

; <<>> DiG 9.18.28-0ubuntu0.22.04.1-Ubuntu <<>> api.rapiddevgroup.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 16659
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;api.rapiddevgroup.com.		IN	A

;; ANSWER SECTION:
api.rapiddevgroup.com.	300	IN	A	172.67.214.239
api.rapiddevgroup.com.	300	IN	A	104.21.91.99

;; Query time: 1208 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Fri Aug 23 23:39:05 CEST 2024
;; MSG SIZE  rcvd: 82
```

Were you able to find the culprit?

**Comment by MarcGodard** [comment]
Yes it is. Like I said, it might be a limitation of pagespeed. If others could test their site and see if they get the error, that would be helpful to figure out if this is a setting causing it, or page speed.

**Comment by HumboldtK** [comment]
![98fb1bc36529c08611877b0f6b64a8ae](https://github.com/user-attachments/assets/a51e9106-6106-4afe-b4cd-aae6d80c7926)

pagespeed gives me the same error as well. My url is resolvable too. I do require a socket connection before my main content is displayed on my site and pagespeed does show it in the preview which mean it connects. Haven't been able to get approved for adsense for this reason too.

**Comment by lethalgerard1** [comment]
https://impressivesingle-spa79.github.io/lkImfGXVIrmIfMznS9CZPYG059LFPii6bDwi4IshoIryMnvHKQphxrOnl68pCII/



The answer just fell into place

**Comment by lethalgerard1** [comment]
https://impressivesingle-spa79.github.io/lkImfGXVIrmIfMznS9CZPYG059LFPii6bDwi4IshoIryMnvHKQphxrOnl68pCII/



The answer just fell into place

**Comment by darrachequesne** [comment]
I think this can be closed now. Please reopen if needed.
